src/pages/index.js

```diff
@@ -77,35 +104,65 @@ function getCardElement(data) {
   const cardLikeBtn = cardElement.querySelector(".card__like-button");
   const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
 
+  // Set card details
   cardNameEl.textContent = data.name;
   cardImageEl.src = data.link;
   cardImageEl.alt = data.name;
 
-  cardLikeBtn.addEventListener("click", () => {
-    cardLikeBtn.classList.toggle("card__like-button_liked");
-  });
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-button_liked");
+  }
 
-  cardImageEl.addEventListener("click", () => {
-    openModal(previewModal);
-    previewModalImageEl.src = data.link;
+  function handleLike(evt, id) {
+    const likeButton = evt.target;
+    const isLiked = likeButton.classList.contains("card__like-button_liked");
+    api
+      .changeLikeStatus(id, isLiked)
+      .then(() => {
+        likeButton.classList.toggle("card__like-button_liked"); 
+      })
+      .catch(console.error);
+  }
+
+function handleImagePreview(data) {
+  previewModalImageEl.src = data.link;
     previewModalCaptionEl.textContent = data.name;
     previewModalImageEl.alt = data.name;
-  });
```

Comment:
You need to format the markup (and code) to show **indentation** and to make the code more readable. You can do that with `Prettier` or a combination `SHIFT + ALT + F`


```jsx

  function handleImagePreview(data) {
    previewModalImageEl.src = data.link;
    previewModalCaptionEl.textContent = data.name;
    previewModalImageEl.alt = data.name;
    openModal(previewModal);
  }
```